mava/components/jax/training/trainer.py

```diff
@@ -131,6 +134,23 @@ def on_building_init_end(self, builder: SystemBuilder) -> None:
         # TODO (Matthew): networks need to be created on the nodes instead?
         builder.store.networks = builder.store.network_factory()
 
+        # Wrap opt_states in a mutable type (dict) since optax return an immutable tuple
```

Comment:
Why is this code duplicated three times? See the other comments.

Reply:
In general, most of the code is repeated so I refactored Trainer.py to have most of the functionality in the Base Class

Reply:
Great :raised_hands: